frontend/src/components/Buttons/FilterButton.vue

```diff
@@ -1,324 +1,299 @@
-          name: ItemFilter.IsResumable
-        },
-        {
-          label: this.$t('favorite'),
-          name: ItemFilter.IsFavorite
-        },
-        { label: this.$t('liked'), name: ItemFilter.Likes },
-        { label: this.$t('unliked'), name: ItemFilter.Dislikes }
-      ],
-      selectedStatusFilters: [],
-      featureFilters: [
-        {
-          label: this.$t('subtitles'),
-          name: 'HasSubtitles'
-        },
-        { label: this.$t('trailer'), value: 'hasTrailer' },
-        {
-          label: this.$t('specialFeatures'),
-          name: 'HasSpecialFeature'
-        },
-        {
-          label: this.$t('themeSong'),
-          name: 'HasThemeSong'
-        },
-        {
-          label: this.$t('themeVideo'),
-          name: 'HasThemeVideo'
-        }
-      ],
-      selectedFeatureFilters: [],
-      genreFilters: [] as string[],
-      selectedGenreFilters: [],
-      ratingFilters: [] as string[],
-      selectedRatingFilters: [],
-      typeFilters: [
-        { label: 'Blu-Ray', name: 'Bluray' },
-        { label: 'DVD', name: 'Dvd' },
-        { label: 'HD', name: 'isHD' },
-        { label: '4K', name: 'is4K' },
-        { label: '3D', name: 'is3D' }
-      ],
-      selectedTypeFilters: [],
-      yearFilters: [] as number[],
-      selectedYearFilters: []
-    };
+  disabled: {
+    type: Boolean,
+    required: false,
+    default: false
+  }
+});
+const statusFilters = [
+  {
+    label: t('played'),
+    name: ItemFilter.IsPlayed
   },
-  watch: {
-    itemsType(): void {
-      this.refreshItems();
-    }
+  {
+    label: t('unplayed'),
+    name: ItemFilter.IsUnplayed
+  },
+  {
+    label: t('resumable'),
+    name: ItemFilter.IsResumable
+  },
+  {
+    label: t('favorite'),
+    name: ItemFilter.IsFavorite
   },
-  methods: {
-    async refreshItems(): Promise<void> {
-      try {
-        /**
-         * Sanitization of route params to avoid XSS injection attacks.
-         *
-         * First reported on https://github.com/jellyfin/jellyfin-vue/security/code-scanning/223
-         */
-        const response = (
-          await this.$remote.sdk
-            .newUserApi(getFilterApi)
-            .getQueryFiltersLegacy({
-              userId: this.$remote.auth.currentUserId.value,
-              parentId: sanitizeHtml(this.$route.params.viewId),
-              includeItemTypes: [this.itemsType]
-            })
-        ).data;
+  { label: t('liked'), name: ItemFilter.Likes },
+  { label: t('unliked'), name: ItemFilter.Dislikes }
+];
+var selectedStatusFilters = ref([]);
+const featureFilters = [
+  {
+    label: t('subtitles'),
+    name: 'HasSubtitles'
+  },
+  { label: t('trailer'), value: 'hasTrailer' },
+  {
+    label: t('specialFeatures'),
+    name: 'HasSpecialFeature'
+  },
+  {
+    label: t('themeSong'),
+    name: 'HasThemeSong'
+  },
+  {
+    label: t('themeVideo'),
+    name: 'HasThemeVideo'
+  }
+];
+const selectedFeatureFilters = ref([]);
+var genreFilters = ref([]);
```

Comment:
refs must be always be consts. Verify this for all the occurences. Do you have IDE warnings?